presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java

```diff
@@ -243,24 +238,6 @@ protected RelationPlan visitJoin(Join node, Void context)
             for (Expression conjunct : ExpressionUtils.extractConjuncts(criteria)) {
                 conjunct = ExpressionUtils.normalize(conjunct);
 
-                // We skip this section for joins using "USING" clause as generated criteria uses non qualified column names.
```

Comment:
I agree 100%. I just wanted this PR to be bugfix rather than refactor. I will do the refactor as last commit in this PR so we can decide if want this in before next release or after.